kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,395 @@
+
+/** \brief   Read data from the flashrom.
+    \ingroup system_calls
+
+    This function reads data from an offset into the flashrom to the 
+    destination buffer.
+
+    \param  pos             The read start position into the flashrom.
+    \param  dest            The destination buffer.
+    \param  n               The number of bytes to read.
+
+    \return                 Number of bytes read on success, or -1 on
+                            failure.
+
+    \sa syscall_flashrom_write(), syscall_flashrom_delete()
+*/
+int syscall_flashrom_read(uint32_t pos, void *dest, size_t n);
+
+/** \brief   Write data to the flashrom.
+    \ingroup system_calls
+
+    This function writes data to an offset into the flashrom from the 
+    source buffer.
+
+    \warning
+    It is only possible to overwrite 1's with 0's. 0's can not be written 
+    back to 1's so general overwriting is therefore not possible.  You 
+    would need to delete a whole partition to overwrite it.
+
+    \param  pos             The start position to write into the flashrom.
+    \param  src             The source buffer.
+    \param  n               The number of bytes to write.
+
+    \return                 Number of bytes written on success, or -1 on
+                            failure.
+
+    \sa syscall_flashrom_read(), syscall_flashrom_delete()
+*/
+int syscall_flashrom_write(uint32_t pos, const void *src, size_t n);
+
+/** \brief   Delete a partition of the flashrom.
+    \ingroup system_calls
+
+    This function returns a flashrom partition to all 1's, so that it may 
+    be rewritten.
+
+    \warning
+    ALL data in the entire partition will be lost.
+
+    \param  pos             The offset from the start of the flashrom you 
+                            want to delete.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+
+    \sa syscall_flashrom_read(), syscall_flashrom_write()
+*/
+int syscall_flashrom_delete(uint32_t pos);
+
+/** \brief   Initialize the GDROM drive.
+    \ingroup system_calls
+
+    This function initializes the GDROM drive. Should be called before any 
+    commands are sent.
+*/
+void syscall_gdrom_init(void);
+
+/** \brief   Reset the GDROM drive.
+    \ingroup system_calls
+
+    This function resets the GDROM drive.
+*/
+void syscall_gdrom_reset(void);
+
+/** \brief   Checks the GDROM drive status.
+    \ingroup system_calls
+
+    This function retrieves the general condition of the GDROM drive. It 
+    populates a provided array with two elements. The first element 
+    indicates the current drive status, and the second element identifies 
+    the type of disk inserted (if any).
+
+    \param  status          The pointer to two 32-bit integers to 
+                            receive extended status information.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_check_drive(void *status);
```

Comment:
```suggestion
int syscall_gdrom_check_drive(int32_t status[2]);
```
There's no need to be ghetto with the typing here, if you literally just said it needs to be a pointer to two 32-bit ints...

But is there a status code enum or any documentation or anything for this anywhere? Where is this used? What is the status? Isn't clear to the user what to look for in the Doxygen to even know what to do with this status.